internal/process/provisioning/resolve_creds_hap_selection_migration_test.go

```diff
@@ -0,0 +1,226 @@
+package provisioning
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/kyma-project/kyma-environment-broker/common/gardener"
+	pkg "github.com/kyma-project/kyma-environment-broker/common/runtime"
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/broker"
+	"github.com/kyma-project/kyma-environment-broker/internal/fixture"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/kyma-project/kyma-environment-broker/common/hyperscaler"
+	"github.com/kyma-project/kyma-environment-broker/common/hyperscaler/rules"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/stretchr/testify/assert"
+
+	k8sTesting "k8s.io/client-go/testing"
+)
+
+// To generate output hap-old-implementation files (used to compare results by hand) switch writeFiles to true
+func TestResolveCredentials_IntegrationAzure2(t *testing.T) {
+	writeFiles := false
+
+	memoryStorage := storage.NewMemoryStorage()
+
+	unstructuredList := &unstructured.UnstructuredList{
+		Items: []unstructured.Unstructured{},
+	}
+
+	var oldImplementationLog *bufio.Writer
+	var newImplementationLog *bufio.Writer
+
+	if writeFiles {
+		oldLog, err := os.OpenFile("./hap-old-implementation.log", os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+		require.NoError(t, err)
+		defer oldLog.Close()
+
+		oldImplementationLog = bufio.NewWriter(oldLog)
+
+		newLog, err := os.OpenFile("./hap-new-implementation.log", os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
```

Comment:
executable? Is this really needed?

Reply:
https://github.com/ralikio/kyma-environment-broker/commit/8b3ca99d045c4f856c12bce4ceee95e506a3c470